Client/DeviceCheck/DeviceCheck.swift

```diff
@@ -0,0 +1,367 @@
+          
+        case .failure(let error):
+          completion(nil, error)
+        }
+      }
+    } catch {
+      completion(nil, error)
+    }
+  }
+  
+  // Sends the attestation to the server along with the nonce and the challenge signature
+  public func setAttestation(nonce: String, _ completion: @escaping (Error?) -> Void) {
+    do {
+      guard let privateKey = try Cryptography.getExistingKey(id: DeviceCheckClient.privateKeyId) else {
+        throw CryptographyError(code: -1, description: "Unable to retrieve existing private key")
+      }
+      
+      let signature = try privateKey.sign(message: nonce).base64EncodedString()
+      let verification = AttestationVerifcation(attestationBlob: AttestationBlob(nonce: nonce),
+                                                signature: signature)
+      
+      try executeRequest(.setAttestation(nonce, verification)) { (result: Result<Data, Error>) in
+        if case .failure(let error) = result {
+          return completion(error)
+        }
+        
+        completion(nil)
+      }
+    } catch {
+      completion(error)
+    }
+  }
+  
+  // Generates a device-check token
+  public func generateToken(_ completion: @escaping (String, Error?) -> Void) {
+        DCDevice.current.generateToken { data, error in
+          if let error = error {
+            return completion("", error)
+          }
+          
+          guard let deviceCheckToken = data?.base64EncodedString() else {
+            return completion("", error)
+          }
+          
+          completion(deviceCheckToken, nil)
+    }
+  }
+  
+  // Generates an enrollment structure to be used with `registerDevice`
+  public func generateEnrollment(paymentId: String, token: String, _ completion: (DeviceCheckRegistration?, Error?) -> Void) {
+    do {
+      guard let privateKey = try Cryptography.generateKey(id: DeviceCheckClient.privateKeyId) else {
+        throw CryptographyError(code: -1, description: "Unable to generate private key")
+      }
+      
+      guard let publicKey = try privateKey.getPublicKeyExternalRepresentationAsPEM() else {
+        throw CryptographyError(code: -1, description: "Unable to retrieve public key")
+      }
+      
+      let signature = try privateKey.sign(message: publicKey + token).base64EncodedString()
+      
+      let enrollmentBlob = DeviceCheckEnrollment(paymentID: paymentId,
+                                                       publicKey: publicKey,
+                                                       deviceToken: token)
+      let registration = DeviceCheckRegistration(enrollmentBlob: enrollmentBlob,
+                                                       signature: signature)
+      completion(registration, nil)
+    } catch {
+      completion(nil, error)
+    }
+  }
+}
+
+private extension DeviceCheckClient {
+  // The base URL of the server
+  private static let baseURL = URL(string: "http://reputation-alb-1044480833.us-west-2.elb.amazonaws.com")
```

Comment:
https

Reply:
![image](https://user-images.githubusercontent.com/1530031/67440929-da862700-f5c8-11e9-857a-8ae07f16ce7f.png)

If I use `https`, I get an error atm. I'm assuming it's because it's dev which is why I never mentioned it and I'm thinking production and staging will be different than dev.